### PR TITLE
bf(S3C-3255): Ensure ingested events timestamp precision

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

**Review by anurag4DSB — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]
LGTM - seems unit test might need adjustment though

File: libV2/server/API/metrics/ingestMetric.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 const errors = require('../../../errors');
 const { UtapiMetric } = require('../../../models');
 const { client: cacheClient } = require('../../../cache');
+const { convertTimestamp } = require('../../../utils');
 
 async function ingestMetric(ctx, params) {
     let metrics;
     try {
-        metrics = params.body.map(m => new UtapiMetric(m));
+        metrics = params.body.map(m => new UtapiMetric({
```
Comment (miniscruff) [inline libV2/server/API/metrics/ingestMetric.js:9]: What about having the UtapiMetric constructor do this itself?

Comment (tmacro) [reply libV2/server/API/metrics/ingestMetric.js:9]: I don't think it's a good idea to have the constructor of a model change the passed data.

Comment (miniscruff) [reply libV2/server/API/metrics/ingestMetric.js:9]: Oh it's not a basic class its the custom JOI model thing I reviewed a while back... then yea ignore me thats not a good idea.

**Review by miniscruff — APPROVED** [review]

**Pull request merged**